openapi/components/schemas/Transactions/CreditTransactionRequest.yaml

```
@@ -1,8 +1,111 @@
-allOf:
-  - $ref: ../Common/CommonTransactionRequest.yaml
-  - properties:
-      payoutRequestId:
-        description: ID of the payout request for which the transaction is fully allocated.
-        writeOnly: true
-        type: string
-        maxLength: 50
+type: object
+required:
+  - websiteId
+  - customerId
+  - currency
+  - amount
+properties:
+  websiteId:
+    $ref: ../Websites/WebsiteId.yaml
+  customerId:
+    $ref: ../CustomerId.yaml
+  currency:
+    $ref: ../CurrencyCode.yaml
+  amount:
+    description: Amount of the transaction.
+    type: number
+    format: double
+    example: 97.97
+  invoiceIds:
+    description: Array of invoice IDs.
+    nullable: true
+    type: array
+    items:
+      $ref: ../ResourceId.yaml
+  paymentInstruction:
+    description: |-
+      Payment instruction for the purchase.
+      If this value is not supplied,
+      the customer's default payment instrument is used.
+    allOf:
+      - $ref: ../PaymentInstruction.yaml
+  billingAddress:
+    description: |-
+      Billing address.
+      If this value is not supplied,
+      the billing address associated with the payment instrument is used.
+      If no billing address is associated with the payment instrument,
+      the customer's billing address is used.
+    nullable: true
+    type: object
+    allOf:
+      - $ref: ../Contact/ContactObject.yaml
+  requestId:
+    description: |-
+      Use this field to prevent duplicate transaction requests that may occur within a short period of time.
+      If a duplicate request is sent with the same `requestId`,
+      it is ignored to prevent double-billing.
+      This value must be unique within a 24-hour period.
+
+      > **Important:** This field is recommended.
+    type: string
+    nullable: true
+    maxLength: 50
+    pattern: '^[\-\w]+$'
+    example: 44433322-2c4y-483z-a0a9-158621f77a21
+  gatewayAccountId:
+    description: |-
+      Rebilly selects the appropriate payment gateway account for the transaction based on the properties of the transaction and the `gateway-account-requested` event rules configurations.
```

suggestion: suggestion for readability 
```suggestion
      ID of the gateway account.
      Rebilly selects the appropriate payment gateway account for the transaction based on the properties of the transaction and the `gateway-account-requested` event rules configurations.
```